op-dispute-mon/mon/extract/withdrawals_enricher.go

```diff
@@ -0,0 +1,49 @@
+package extract
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/contracts"
+	monTypes "github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum-optimism/optimism/op-service/sources/batching/rpcblock"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/exp/maps"
+)
+
+var ErrIncorrectWithdrawalsCount = errors.New("incorrect withdrawals count")
+
+var _ Enricher = (*WithdrawalsEnricher)(nil)
+
+type WithdrawalsEnricher struct{}
+
+func NewWithdrawalsEnricher() *WithdrawalsEnricher {
+	return &WithdrawalsEnricher{}
+}
+
+func (w *WithdrawalsEnricher) Enrich(ctx context.Context, block rpcblock.Block, _ GameCaller, caller WethCaller, game *monTypes.EnrichedGameData) error {
+	recipients := make(map[common.Address]bool)
+	for _, claim := range game.Claims {
+		if claim.CounteredBy != (common.Address{}) {
+			recipients[claim.CounteredBy] = true
+		} else {
+			recipients[claim.Claimant] = true
+		}
+	}
+	recipientAddrs := maps.Keys(recipients)
```

Comment:
I feel like we've implemented this a number of times in different places.  Maybe we should add it as its own enricher and just store the map of recipients on `EnrichedGameData?  Could be a follow up PR though.